includes/admin/class-wc-rest-payments-terminal-locations-controller.php

```diff
@@ -125,4 +175,107 @@ private function format_location_response( $location ) {
 			]
 		);
 	}
+
+	/**
+	 * Proxies the delete location request to the server.
+	 *
+	 * @param WP_REST_REQUEST $request Request object.
+	 *
+	 * @throws API_Exception - If the downstream call fails.
+	 */
+	public function delete_location( $request ) {
+		try {
+			$deletion_response = $this->api_client->delete_terminal_location(
+				$request->get_param( 'location_id' )
+			);
+
+			// Delete the transient in case the delete call goes through to avoid caching side effects.
+			if ( true === $deletion_response['deleted'] ?? false ) {
+				delete_transient( self::STORE_LOCATIONS_TRANSIENT_KEY );
+			}
+
+			return $deletion_response;
+		} catch ( API_Exception $e ) {
+			return rest_ensure_response( new WP_Error( $e->get_error_code(), $e->getMessage() ) );
+		}
+
+	}
+
+	/**
+	 * Proxies the update location request to the server.
+	 *
+	 * @param WP_REST_REQUEST $request Request object.
+	 *
+	 * @throws API_Exception - If the downstream call fails.
+	 */
+	public function update_location( $request ) {
+		try {
+			$location_id  = $request->get_param( 'location_id' );
+			$display_name = $request['display_name'];
+			$address      = $request['address'];
+
+			$updation_response = $this->api_client->update_terminal_location( $location_id, $display_name, $address );
+
+			// Delete the transient in case the update call goes through to avoid caching side effects.
+			delete_transient( self::STORE_LOCATIONS_TRANSIENT_KEY );
+
+			return $this->format_location_response( $updation_response );
+		} catch ( API_Exception $e ) {
+			return rest_ensure_response( new WP_Error( $e->get_error_code(), $e->getMessage() ) );
+		}
+	}
+
+	/**
+	 * Proxies the get location request to the server.
+	 *
+	 * @param WP_REST_REQUEST $request Request object.
+	 *
+	 * @throws API_Exception - If the downstream call fails.
+	 */
+	public function get_location( $request ) {
+		try {
+			$locations   = get_transient( static::STORE_LOCATIONS_TRANSIENT_KEY );
+			$location_id = $request->get_param( 'location_id' );
+
+			// Check if the location exists in cache before making the request.
+			if ( $locations ) {
+				foreach ( $locations as $location ) {
+					if ( $location['id'] === $location_id ) {
+						return $this->format_location_response( $location );
+					}
+				}
+			}
+
+			return $this->format_location_response(
+				$this->api_client->get_terminal_location( $location_id )
```

Comment:
Right now transient is only updated in the store location endpoint. Given that we drop transients on update/delete I would expect us to retrieve all locations here and update transient so that we can use an up-to-date cached data for any follow-up requests, similar as it happens on lines 134-137.

Reply:
Great catch. We try to check if the specific location exists in transient and serve it from there before making the API call in the block above so yeah, updating transient to include this fetched one in case of a cache miss makes sense for follow-up requests - will make this change.

Reply:
Addressed in [9026ae6](https://github.com/Automattic/woocommerce-payments/pull/3013/commits/9026ae665b269f0ec7e5d3163552d940c38f2533).